### PR TITLE
Reuse markdig pipeline using InclusionContext

### PR DESCRIPTION
**Pull request by yufeih**

### CODE REVIEW

**Comment by OsmondJiang** [comment]
> // Licensed under the MIT license. See LICENSE file in the project root for full license information.

...., this PR's change is too big...

---
Refers to: src/Microsoft.DocAsCode.MarkdigEngine/MarkdigMarkdownService.cs:2 in bfa9ef4. [](commit_id = bfa9ef438c936ae19f01f05dec21a42e48e1dcdc, deletion_comment = False)

File: test/Microsoft.DocAsCode.MarkdigEngine.Tests/MarkdigServiceTest.cs [hunk 1]
```diff
@@ -64,19 +64,5 @@ public void MarkdigServiceTest_ParseAndRender_Inclusion()
             var expectedDependency = new List<string> { "~/x/b/linkAndRefRoot.md" };
             Assert.Equal(expectedDependency.ToImmutableList(), mr.Dependency);
         }
-
-        [Fact]
-        [Trait("Related", "MarkdigService")]
-        public void MarkdigServiceTest_ParseInline()
```
Comment (qinezh) [inline test/Microsoft.DocAsCode.MarkdigEngine.Tests/MarkdigServiceTest.cs]: >MarkdigServiceTest_ParseInline [](start = 20, length = 30)

why remove this test case?

Comment (yufeih) [reply test/Microsoft.DocAsCode.MarkdigEngine.Tests/MarkdigServiceTest.cs]: I'll add back the `isInline` interface and revert the change

**Comment by yufeih** [comment]
I'll create several smaller ones

**Pull request closed without merging**